job/job.go

```diff
@@ -248,9 +248,41 @@ func (j *Job) Disable() {
 	j.Disabled = true
 }
 
+// DeleteFromParentJobs goes through and deletes the current job from any parent jobs.
+func (j *Job) DeleteFromParentJobs(cache JobCache) error {
+	j.lock.Lock()
+	defer j.lock.Unlock()
+	if len(j.ParentJobs) != 0 {
+		for _, p := range j.ParentJobs {
+			parentJob, err := cache.Get(p)
+			parentJob.lock.Lock()
```

Comment:
@ajvb This is not safe; if there's an error then parentJob will be nil and this line will panic.